.azure-pipelines/gpg/activate-key.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash -eu
+
+set -o pipefail
+
+GNUPGHOME="${GNUPGHOME:-${HOME}/.gnupg}"
+GPG_PASSPHRASE="${GPG_PASSPHRASE:-HACKME}"
+GPG_KEYFILE="${GPG_KEYFILE:-test.env/ci.snakeoil.gpg.key}"
+
+# Azp leaves non existent vars as they are (???)
+#
+# shellcheck disable=SC2016
+if [[ "$GPG_KEYFILE" == '$(MaintainerGPGKey.secureFilePath)' ]]; then
+    GPG_PASSPHRASE=HACKME
+    GPG_KEYFILE="test.env/ci.snakeoil.gpg.key"
+
+fi
```

Comment:
Sorry if missed something. I didn't get this, why? 🤣 I think the whole flow is like this:
- generate test key in the env
- download test key or maintainer key based on the test type (postsubmit or not)
- load the key (test key or maintainer key)

IMO, the GPG_KEYFILE should be set to the path of downloaded key and it may be different for maintainer key and test key.
However, the script always set the GPG_KEYFILE to the `parameters.authGPGPath`(`MaintainerGPGKey.secureFilePath`).
So, I am a little confused now.

Reply:
I guess the `MaintainerGPGKey.secureFilePath` is configured in the cloud and is the path to store the downloaded maintainer key?

Reply:
>  the GPG_KEYFILE should be set to the path of downloaded key and it may be different for maintainer key and test key.

yes that is what its intended to do 

> the script always set the GPG_KEYFILE to the  ... `MaintainerGPGKey.secureFilePath`

this only exists in postsubmit 

in presubmit its best to think of it as an empty string - any rational language would have done that - in azp its the original string ie `$(MaintainerGPGKey.secureFilePath)`

so instead of just testing whether its set - we have to test whether its set to the uninterpolated string

Reply:
> I guess the MaintainerGPGKey.secureFilePath is configured in the cloud and is the path to store the downloaded maintainer key?

yep its an azp secure file

https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/download-secure-file-v1?view=azure-pipelines

Reply:
> in presubmit its best to think of it as an empty string - any rational language would have done that - in azp its the original string ie 

wt......, nightware.

Could we document this more clear then other readers could get it?

Reply:
> Could we document this more clear then other readers could get it?

yep let me think how to do that - i added the "non existent vars" comment but probs that is not sufficient explanation

Reply:
updated, hopefully its a bit clearer